### ISSUE TITLE
Error on duplicate firewall rules - no way of checking the error code

### ISSUE THREAD

**Issue by FireFart** [cloudflare/cloudflare-go#574]
Currently we try to do automated blocking on cloudflare by creating IP access rules.
Example:
```golang
resp, err := c.client.CreateAccountAccessRule(accountID, cloudflare.AccessRule{
		Mode:  "block",
		Notes: comment,
		Configuration: cloudflare.AccessRuleConfiguration{
			Target: target,
			Value:  ip,
		},
	})
```

The problem is this rules takes some time to propagate and if we try to block the same ip again, the server returns a HTTP 400 with the following body

```json
{
  "result": null,
  "success": false,
  "errors": [
    {
      "code": 10009,
      "message": "firewallaccessrules.api.duplicate_of_existing"
    }
  ],
  "messages": []
}
```

## Expected behavior

I would expect that the `Response` object returned by the client contains `Success` = false and the error in the `Error` object

## Current behavior

An error is returned as the HTTP status is 400 with the following content:
```
error from makeRequest: HTTP status 400: content "{\n  \"result\": null,\n  \"success\": false,\n  \"errors\": [\n    {\n      \"code\": 10009,\n      \"message\": \"firewallaccessrules.api.duplicate_of_existing\"\n    }\n  ],\n  \"messages\": []\n}\n"
```

https://github.com/cloudflare/cloudflare-go/blob/a71461723e375f9fb4126b5d973b2619bb010f27/cloudflare.go#L281

This makes it really hard to check the returned error code. I think it would be best to either return a custom error with the error information that can be checked using `errors.Is` or `errors.As` or the response object is properly populated (but this might cause some errors on existing implementations when people rely on an returned error instead of checking the `Success` flag)

**Comment by jacobbednarz** [comment]
Totally agree this is harder than it needs to be in some cases. It's definitely on my radar to address now that we have better error handling available in Go.

For now, I recommend you use something like `strings.Contains()` and check for a known bad attribute such as `\"success\": false` or the code to accomplish the outcome you need.

```go
if err != nil {
		// check for known error codes as strings; otherwise retry the request
		if strings.Contains(err.Error(), "\\\"code\\\": 10009") || strings.Contains(err.Error(), "firewallaccessrules.api.duplicate_of_existing") {
			// skip the operation or accept you've got a duplicate
		}
	}
```

**Comment by jacobbednarz** [comment]
released in [v0.14.0](https://github.com/cloudflare/cloudflare-go/releases/tag/v0.14.0)

**Comment by FireFart** [comment]
awesome thanks!
